adaptive/tests/test_average_learner.py

```diff
@@ -46,3 +47,20 @@ def test_avg_std_and_npoints():
             assert learner.npoints == len(learner.data)
             assert abs(learner.sum_f - values.sum()) < 1e-13
             assert abs(learner.std - std) < 1e-13
+
+
+def test_min_npoints():
```

Comment:
The test seems unnecessarily complex. I propose to use a fully constant function and check the number of points when done.

Or perhaps I don't understand why `npoints_similar` is relevant for the testing behavior.

Reply:
I simplified the test in https://github.com/python-adaptive/adaptive/pull/274/commits/02c4047fc705fb3b79a16e5b9cb2fd98897babd1. Good?

Reply:
It could be even simpler with a constant function (like you said). Although I am not sure whether a reader of this test would understand the underlying problem that it's testing.

Reply:
Let's simplify this further.

- We don't need actual randomness.
- We need to ensure that a function with which the learner would stop before reaching min_points runs for at least min_points. Therefore we need to assert that if we run without `min_points`, then 2 points are evaluated, and otherwise at least `min_points.